spec/puppet-server-lib/puppet/jvm/ast_compiler_spec.rb

```diff
@@ -70,5 +75,70 @@ def find_notify(catalog)
         Puppet::Server::ASTCompiler.compile(bolt_request, boltlib_path)
       }.to raise_error(Puppet::Error, /the path to boltlib modules must be provided/)
     end
+
+    context 'when compiling for a plan with fact and plan/target variable collisions' do
+      let(:boltlib_path) { [] }
+
+      it 'properly shadows the relevant variables' do
+        # TODO: This is a temporary stub of the Bolt requires to get this
+        # test to pass. We should figure out how to properly load Bolt code
+        # later for the tests.
+        allow(Puppet::Server::ASTCompiler).to receive(:require).and_call_original
```

Comment:
I know this sucks, but unless there's a _very_ good reason for us not to mock `require` in the tests, I'd rather not spend anymore time figuring out how to handle the Bolt requires in the AST compiler.

Reply:
I briefly looked at this. When i did found that it was actually a pretty small subset of all the bolt dependency gems that we actually would need. I think so far we have relied heavily on testing this in orchestrator acceptance tests.

Reply:
Yeah, I'm going to add Orch acceptance tests for the shadowing if none currently exist. These tests at least are an easy way for us to detect a regression there if it happens so still useful to have even if the mocking currently sucks.

Reply:
How about we pull out the `require`s into a single `require_bolt` method we can stub normally?

Reply:
works for me. I'll make that update. good idea as usual @nicklewis :100: